### ISSUE TITLE
@Lazy decorator?

### ISSUE THREAD

**Issue by jarrvis** [ionic-team/stencil#1341]
**Stencil version:**

```
 @stencil/core@<0.17.0>
```

**I'm submitting a:**

[ ] bug report
[x] feature request
[ ] support request => Please do not submit support requests here, use one of these channels: https://stencil-worldwide.herokuapp.com/ or https://forum.ionicframework.com/

**Expected behavior:**
Hi,
so I was checking this nice repo https://github.com/jgw96/st-img and it got me thinking... 
What do you think about introducing @Lazy decorator that would call annotated method after you scroll to component?
Possible usage: I have a component that requires data from API for instance 'recently viewed products' component - carousel on the bottom of the page. I want to feed this component (paying with http request) only when user is actually on the bottom of the page. Obviously I would feed the component only once.

**Comment by manucorporat** [comment]
We don't have plans to extend the API surface of stencil for the moment! This seems like an feature that could be implemented as an external library, not part of the core!

**Comment by jarrvis** [comment]
And so it was: https://www.npmjs.com/package/st-lazy
